asyncscope.md

```diff
@@ -2317,7 +2317,244 @@ spec here
 
 ## `execution::simple_counting_scope`
 
-spec here
+Add the following new section imediate after **TODO**:
+
+::: add
+### Scopes [exec.scopes]
+
+#### Simple Counting Scope [exec.simple.counting.scope]
+
+##### General [exec.scounting.general]
+
+```
+class simple_counting_scope {
+public:
+  // [exec.scounting.token], token
+  struct token {
+    template <sender Sender>
+    Sender&& wrap(Sender&& snd) const noexcept;
+    assoc try_associate() const;
+
+  private:
+    simple_counting_scope* @_scope_@; // @_exposition-only_@
+  };
+
+  // [exec.scounting.assoc], assoc
+  struct assoc {
+    assoc() noexcept = default;
+    assoc(const assoc& other) noexcept;
+    assoc(assoc&& other) noexcept;
+    ~assoc();
+    assoc& operator=(assoc rhs) noexcept;
+
+    explicit operator bool() const noexcept;
+
+  private:
+    simple_counting_scope* @_scope_@{}; // @_exposition-only_@
+  };
+
+  struct @_join-t_@; // @_exposition-only_@
+
+  enum @_state-type_@ { // @_exposition-only_@
+    @_unused_@, // @_exposition-only_@
+    @_open_@, // @_exposition-only_@
+    @_close_@, // @_exposition-only_@
+    @_open-and-joining_@, // @_exposition-only_@
+    @_closed-and-joining_@, // @_exposition-only_@
+    @_unused-and-closed_@, // @_exposition-only_@
+    @_joined_@, // @_exposition-only_@
+  };
+
+  // [exec.scounting.ctor], constructor and destructor
+  simple_counting_scope() noexcept;
+  simple_counting_scope(simple_counting_scope&&) = delete;
+  ~simple_counting_scope();
+
+  // [exec.scounting.mem], members
+  token get_token() noexcept;
+  void close() noexcept;
+  auto join() noexcept;
```

Comment:
Is it worth constraining this `auto`?
```suggestion
  sender auto join() noexcept;
```

Reply:
I made it consistent with the class definition. For `counting_scope` the member `join()` is specified to return a `join-sender`: maybe that should be done here as well?